sphinx/transforms/post_transforms/__init__.py

```diff
@@ -197,11 +200,71 @@ def apply(self):
                     node.replace_self(nodes.comment())
 
 
+class SphinxSmartQuotes(SmartQuotes, SphinxTransform):
```

Comment:
Ah, I noticed post_transforms does not effect to toctree and other data. As a result, some testcases are failed now. I need to refactor this more and more.

Reply:
In addition, the document object generated in `SphinxTransformer` does not have `settings.smart_quotes` attribute. Because it does not use `RSTParser` on instantiation.
As a result, some tests are failed (ex `test_ext_todo.py`).